src/clib/pio_darray.c

```diff
@@ -108,7 +108,7 @@ int PIOc_write_darray_multi(int ncid, const int *varids, int ioid, int nvars,
     iosystem_desc_t *ios;  /* Pointer to io system information. */
     file_desc_t *file;     /* Pointer to file information. */
     io_desc_t *iodesc;     /* Pointer to IO description information. */
-    int rlen;              /* Total data buffer size. */
+    size_t rlen;           /* Total data buffer size. */
```

Comment:
In future we might need to take a deeper look at the different types and conversions used in Scorpio, AFAICT it has been used indiscriminately throughout the legacy code. For example, bget() accepts longs, PIO_Offset is a type equivalent to MPI_Offset and we also use size_t here.